resources/fetching/fetchers/azure/batch_fetcher.go

```diff
@@ -57,29 +57,58 @@ func (f *AzureBatchAssetFetcher) Fetch(ctx context.Context, cMetadata fetching.C
 		}
 
 		if len(assets) == 0 {
+			// Hack backported into 8.11:
+			// Get the first subscription (should only be one), get the sub id and name and populate a mock resource
+			// with an empty array.
+			subId, subName := func() (string, string) {
+				for subId, subName := range f.provider.GetSubscriptions() {
+					return subId, subName
+				}
+				return "", ""
+			}()
+
+			success := f.write(ctx, fetching.ResourceInfo{
+				Resource: &EmptyBatchResource{
+					Type:    pair.Type,
+					SubType: pair.SubType,
+					SubId:   subId,
+					SubName: subName,
+				},
+				CycleMetadata: cMetadata,
+			})
+			if !success {
+				return nil
```

Comment:
Maybe if we cannot send a batch we should continue and try to send other batches?
The same goes for other cases of returning nil on errors.
Also the API calls for listing types as well since if we get an error there we essentially skip the whole batch fetcher, but we can at least try to query other batches.

Reply:
This is just for case of context canceled, where we return (as we did before)